src/Components/Categories/CategoriesTable.js

```diff
@@ -0,0 +1,144 @@
+import { format as formatDate, parseISO } from "date-fns";
+import React, { useMemo, useState } from "react";
+import { useTable } from "react-table";
+import "./CategoriesTable.css";
+
+//fontawesome
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { faTrashAlt } from "@fortawesome/free-solid-svg-icons";
+import { faEdit } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import Modal from "./../Elements/Modal";
+import CategoryEditForm from "./../Categories/CategoryEditForm";
+import {
+  updateCategory,
+  deleteCategory,
+} from "../../Services/Organizations/CategoriesService";
+import { useCurrentOrganizationContext } from "../../Contexts/currentOrganizationContext";
+
+library.add(faTrashAlt);
+library.add(faEdit);
+
+export default function CategoriesTable(props) {
+  const renderDateColumn = (dateString) =>
+    formatDate(parseISO(dateString), "PP");
+
+  const [name, setName] = useState("");
+  const [selectedCategory, setSelectedCategory] = useState({});
+
+  const [showCategoryEdit, setShowCategoryEdit] = useState(false);
+  const handleClose = () => {
+    setShowCategoryEdit(false);
+  };
+  const {
+    currentOrganizationStore,
+    currentOrganizationDispatch,
+    organizationClient,
+  } = useCurrentOrganizationContext();
+  const currentOrganizationId =
+    currentOrganizationStore.currentOrganization &&
+    currentOrganizationStore.currentOrganization.id;
+
+  const handleShowCategoryEdit = (selectedCategory) => {
+    setSelectedCategory(selectedCategory);
+    setShowCategoryEdit(true);
+  };
+
+  const handleSubmitEditCategory = ({ newName }, id) => {
+    console.log(id);
+    updateCategory(organizationClient, id, {
+      name: newName,
+      organization_id: currentOrganizationId,
+    })
+      .then((category) => {
+        setName(name);
+        handleClose();
+      })
+      .catch((error) => {
+        console.log("category update error", error);
+      });
```

Comment:
What do you think of moving the API call to the parent to keep the Table focused on rendering the data. Maybe this component could get passed the two functions as props: `onEditCategory`, `onDeleteCategory`.

If we did this, we could also move the modal to the parent.

Reply:
I like this idea.